src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -181,86 +181,445 @@ internal static DateTimeOffset GetUtcDateTimeOffset(DateTime dateTime)
+        /// </exception>
+        /// <remarks>
+        ///     This method may return an inaccurate value, because it uses native functions whose values may not
+        ///     be continuously updated by the operating system.
+        /// </remarks>
         public static DateTime GetLastWriteTimeUtc(SafeFileHandle fileHandle)
-            => FileSystem.GetLastWriteTime(fileHandle).UtcDateTime;
+        {
+            ArgumentNullException.ThrowIfNull(fileHandle);
+            return FileSystem.GetLastWriteTime(fileHandle).UtcDateTime;
+        }
 
         public static FileAttributes GetAttributes(string path)
             => FileSystem.GetAttributes(Path.GetFullPath(path));
 
+        /// <summary>
+        ///     Gets the specified <see cref="FileAttributes"/> of the file or directory by <paramref name="fileHandle"/>.
+        /// </summary>
+        /// <param name="fileHandle">
+        ///     A <see cref="SafeFileHandle" /> to the file or directory for which the attributes are to be retrieved.
+        /// </param>
+        /// <returns>
+        ///     The <see cref="FileAttributes"/> of the file on the path.
+        /// </returns>
+        /// <exception cref="UnauthorizedAccessException">
+        ///     The caller does not have the required permission.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="fileHandle"/> is <see langword="null"/>.
+        /// </exception>
         public static FileAttributes GetAttributes(SafeFileHandle fileHandle)
-            => FileSystem.GetAttributes(fileHandle);
+        {
+            ArgumentNullException.ThrowIfNull(fileHandle);
+            return FileSystem.GetAttributes(fileHandle);
+        }
 
         public static void SetAttributes(string path, FileAttributes fileAttributes)
             => FileSystem.SetAttributes(Path.GetFullPath(path), fileAttributes);
 
+        /// <summary>
+        ///     Sets the specified <see cref="FileAttributes"/> of the file or directory by <paramref name="fileHandle"/>.
+        /// </summary>
+        /// <param name="fileHandle">
+        ///     A <see cref="SafeFileHandle" /> to the file or directory for <paramref name="fileAttributes"/> should be set.
+        /// </param>
+        /// <param name="fileAttributes">
+        ///     A bitwise combination of the enumeration values.
+        /// </param>
+        /// <exception cref="UnauthorizedAccessException">
+        ///     The caller does not have the required permission.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="fileHandle"/> is <see langword="null"/>.
+        /// </exception>
+        /// <remarks>
+        ///    It is not possible to change the compression status of a <see cref="File"/> object
+        ///    using the <see cref="SetAttributes(SafeFileHandle, FileAttributes)"/> method.
+        /// </remarks>
```

Comment:
This is not in the docs for `SetAttributes(string, FileAttributes)`, should it be? What motivated you to add it here?

Reply:
it is on msdn
https://docs.microsoft.com/en-us/dotnet/api/system.io.file.setattributes?view=net-6.0